Services.Tests/ExternalDataProviders/DataProviderLoaderTest.cs

```diff
@@ -0,0 +1,276 @@
+/*
+* Digital Excellence Copyright (C) 2020 Brend Smits
+*
+* This program is free software: you can redistribute it and/or modify
+* it under the terms of the GNU Lesser General Public License as published
+* by the Free Software Foundation version 3 of the License.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty
+* of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+* See the GNU Lesser General Public License for more details.
+*
+* You can find a copy of the GNU Lesser General Public License
+* along with this program, in the LICENSE.md file in the root project directory.
+* If not, see https://www.gnu.org/licenses/lgpl-3.0.txt
+*/
+
+using API.Configuration;
+using AutoMapper;
+using FluentAssertions;
+using Microsoft.Extensions.DependencyInjection;
+using Models;
+using Moq;
+using NUnit.Framework;
+using Repositories;
+using Services.ExternalDataProviders;
+using Services.ExternalDataProviders.Helpers;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Services.Tests.ExternalDataProviders
+{
+
+    [TestFixture]
+    public class DataLoaderTest
+    {
+
+        private IDataProviderLoader loader;
+
+        private Mock<IServiceScopeFactory> factoryMock;
+        private Mock<IDataSourceModelRepository> repositoryMock;
+        private Mock<IAssemblyHelper> assemblyHelperMock;
+        private IMapper mapper;
+
+        private Mock<IDataSourceAdaptee> returnedTypeFromLoader;
+
+        private Type[] returnedTypesFromAssembly;
+
+        [SetUp]
+        public void Initialize()
+        {
+            factoryMock = new Mock<IServiceScopeFactory>();
+
+            repositoryMock = new Mock<IDataSourceModelRepository>();
+
+            assemblyHelperMock = new Mock<IAssemblyHelper>();
+
+            MapperConfiguration mappingConfig = new MapperConfiguration(mc => mc.AddProfile(new MappingProfile()));
+            mapper = mappingConfig.CreateMapper();
+
+            loader = new DataProviderLoader(factoryMock.Object, repositoryMock.Object, assemblyHelperMock.Object ,mapper);
+        }
+
+
+        [Test]
+        public async Task GetAllDataSources_GoodFlow()
```

Comment:
Can you maybe add a description how the test accomplishes the good flow?

Reply:
Also maybe do this for the other tests? Would be nice for rookies like me to understand it! :)